File: Program/c++/Motor.cpp
#include <iostream>
#include "Kendaraan.cpp"

// subclass dari kendaraan
class Motor : public Kendaraan {
    private:
        int kapasitasMesin; // dalam cc
        string jenisTransmisi; // manual / otomatis
        double kapasitasTangki; // dalam liter

    public: 
        // Default Constructor
        Motor() : Kendaraan() {
            this->kapasitasMesin = 0;
            this->jenisTransmisi = "";
            this->kapasitasTangki = 0.0;
        }

        // Constructor dengan Parameter
        Motor(int id, string merk, int tahun, int kapasitasMesin, string jenisTransmisi, double kapasitasTangki) 
            : Kendaraan(id, merk, tahun) {
            this->kapasitasMesin = kapasitasMesin;
            this->jenisTransmisi = jenisTransmisi;
            this->kapasitasTangki = kapasitasTangki;
        }

        // Getter dan Setter Kapasitas Mesin
        void setKapasitasMesin(int kapasitasMesin) { this->kapasitasMesin = kapasitasMesin; }
        int getKapasitasMesin() { return this->kapasitasMesin; }

        // Getter dan Setter Jenis Transmisi
        void setJenisTransmisi(string jenisTransmisi) { this->jenisTransmisi = jenisTransmisi; }
        string getJenisTransmisi() { return this->jenisTransmisi; }

        // Getter dan Setter Kapasitas Tangki
        void setKapasitasTangki(double kapasitasTangki) { this->kapasitasTangki = kapasitasTangki; }
        double getKapasitasTangki() { return this->kapasitasTangki; }

        // Destructor
        ~Motor() {}

};

File: Program/Java/Motor.java
// Subclass dari kendaraan (Inheritance)
public class Motor extends Kendaraan{
    // Atribut motor
    private int kapasitasMesin; // kapasitas mesin dalam cc
    private String jenisTransmisi; // manual/otomatis
    private double kapasitasTangki; // kapasitas tangki bensin dalam Liter

    // Constructor
    public Motor (int id, String merk, int tahun, int kapasitasMesin, String jenisTransmisi, double kapasitasTangki){
        super(id, merk, tahun);
        this.kapasitasMesin = kapasitasMesin;
        this.jenisTransmisi = jenisTransmisi;
        this.kapasitasTangki = kapasitasTangki;
    }

    // Setter dan Getter Kapasitas Mesin
    public void setKapasitasMesin(int kapasitasMesin){this.kapasitasMesin = kapasitasMesin;}
    public int getKapasitasMesin(){return this.kapasitasMesin;}

    // Setter dan Getter jenis transmisi
    public void setJenisTransmisi(String jenisTransmisi){this.jenisTransmisi = jenisTransmisi;}
    public String getJenisTransmisi(){return this.jenisTransmisi;}

    // setter dan getter kapasitas tangki
    public void setKapasitasTangki(double kapasitasTangki){this.kapasitasTangki = kapasitasTangki;}
    public double getKapasitasTangki(){return this.kapasitasTangki;}

}
